tmux/formater.go

```diff
@@ -306,37 +316,61 @@ func (f *Formater) formatFlag(style, symbol string, count int) string {
 func (f *Formater) flags() string {
 	var flags []string
 	if f.st.IsClean {
-		if f.st.NumStashed != 0 && f.Symbols.Stashed != "" {
-			flags = append(flags, f.formatFlag(f.Styles.Stashed, f.Symbols.Stashed, f.st.NumStashed))
+		if f.st.NumStashed != 0 {
+			flag := f.formatFlag(f.Styles.Stashed, f.Symbols.Stashed, f.st.NumStashed)
+			if flag != "" {
+				flags = append(flags, flag)
+			}
 		}
 
-		if !f.Options.HideClean && f.Symbols.Clean != "" {
-			flags = append(flags, fmt.Sprintf("%s%s", f.Styles.Clean, f.Symbols.Clean))
+		if !f.Options.HideClean {
+			// Handle clean symbol separately since it doesn't have a count
+			if f.Symbols.Clean != "" {
+				flags = append(flags, fmt.Sprintf("%s%s", f.Styles.Clean, f.Symbols.Clean))
+			} else if !f.Options.HideFlagCountIfEmptySymbol {
+				// When symbol is empty but we don't want to hide, there's no count to show for clean flag
+				// so we just skip it (nothing meaningful to display)
+			}
 		}
 
 		if len(flags) != 0 {
 			return f.Styles.Clear + strings.Join(flags, " ")
 		}
 	}
 
-	if f.st.NumStaged != 0 && f.Symbols.Staged != "" {
-		flags = append(flags, f.formatFlag(f.Styles.Staged, f.Symbols.Staged, f.st.NumStaged))
+	if f.st.NumStaged != 0 {
+		flag := f.formatFlag(f.Styles.Staged, f.Symbols.Staged, f.st.NumStaged)
+		if flag != "" {
+			flags = append(flags, flag)
+		}
 	}
 
-	if f.st.NumConflicts != 0 && f.Symbols.Conflict != "" {
-		flags = append(flags, f.formatFlag(f.Styles.Conflict, f.Symbols.Conflict, f.st.NumConflicts))
+	if f.st.NumConflicts != 0 {
+		flag := f.formatFlag(f.Styles.Conflict, f.Symbols.Conflict, f.st.NumConflicts)
+		if flag != "" {
+			flags = append(flags, flag)
+		}
 	}
 
-	if f.st.NumModified != 0 && f.Symbols.Modified != "" {
-		flags = append(flags, f.formatFlag(f.Styles.Modified, f.Symbols.Modified, f.st.NumModified))
+	if f.st.NumModified != 0 {
+		flag := f.formatFlag(f.Styles.Modified, f.Symbols.Modified, f.st.NumModified)
+		if flag != "" {
+			flags = append(flags, flag)
+		}
 	}
 
-	if f.st.NumStashed != 0 && f.Symbols.Stashed != "" {
-		flags = append(flags, f.formatFlag(f.Styles.Stashed, f.Symbols.Stashed, f.st.NumStashed))
+	if f.st.NumStashed != 0 {
+		flag := f.formatFlag(f.Styles.Stashed, f.Symbols.Stashed, f.st.NumStashed)
+		if flag != "" {
+			flags = append(flags, flag)
+		}
 	}
 
-	if f.st.NumUntracked != 0 && f.Symbols.Untracked != "" {
-		flags = append(flags, f.formatFlag(f.Styles.Untracked, f.Symbols.Untracked, f.st.NumUntracked))
+	if f.st.NumUntracked != 0 {
+		flag := f.formatFlag(f.Styles.Untracked, f.Symbols.Untracked, f.st.NumUntracked)
+		if flag != "" {
+			flags = append(flags, flag)
+		}
```

Comment:
Code would be much simpler if formatFlag mehod was:

```
func (f *Formater) appendFlag(flags []string, style, symbol string, count int) []string
```

It would be called like that for each flag kind:

```
flags = f.appendFlag(flags, f.Styles.Stashed, f.Symbols.Stashed, f.st.NumStashed)
```

and appendFlag method would take care of handling all cases with both flag-related options